tests/foreman/ui/test_host.py

```diff
@@ -2985,3 +2987,147 @@ def test_positive_manage_repository_sets(
+        content_view=cv,
+        content_source=1,
+    ).create()
+
+    with target_sat.ui_session() as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_location.name)
+        session.computeresource.create(
+            {
+                'name': cr_name,
+                'provider': FOREMAN_PROVIDERS['vmware'],
+                'provider_content.vcenter': vmware.hostname,
+                'provider_content.user': settings.vmware.username,
+                'provider_content.password': settings.vmware.password,
+                'provider_content.datacenter.value': settings.vmware.datacenter,
+                'locations.resources.assigned': [module_location.name],
+                'organizations.resources.assigned': [module_org.name],
+            }
+        )
+        session.hostgroup.update(hostgroup_name, {'host_group.deploy': cr_name + " (VMware)"})
+
+        cr_vm_names = [settings.vmware.vm_name, 'phoenix-testing-guest-rhel-8']
+        vm_names_with_domains = [f'{name.replace(".", "")}.{domain.name}' for name in cr_vm_names]
+
+        # Import VMs from VMware compute resource
+        for cr_vm_name, vm_name_with_domain in zip(
+            cr_vm_names, vm_names_with_domains, strict=False
+        ):
+            session.computeresource.vm_import(
+                cr_name,
+                cr_vm_name,
+                hostgroup_name,
+                module_location.name,
+                name=cr_vm_name.replace('.', ''),
+            )
+            assert session.all_hosts.search(vm_name_with_domain)
+
+        @request.addfinalizer
+        def _cleanup():
+            for vm_name in vm_names_with_domains:
+                try:
+                    target_sat.cli.Host.delete({'name': vm_name})
+                except CLIReturnCodeError as e:
+                    print(f"Failed to delete VM {vm_name}: {e}")
+
+        vm_values_pre_disassociation = {}
+        for vm_name in vm_names_with_domains:
+            # Get info about host from API
+            api_val_pre_disassociation = target_sat.api.Host().search(
+                query={"search": f'name={vm_name}'}
+            )[0]
+            vm_values_pre_disassociation[vm_name] = api_val_pre_disassociation
+            # Check that uuid and compute_resource_id are set
+            assert api_val_pre_disassociation.uuid is not None, f"UUID for {vm_name} is not set"
+            assert api_val_pre_disassociation.compute_resource.id is not None, (
+                f"Compute resource ID for {vm_name} is not set"
+            )
+
+        session.all_hosts.disassociate_hosts(host_names=vm_names_with_domains)
+
+        vm_values_post_disassociation = {}
+        for vm_name in vm_names_with_domains:
+            # Get info about host from API
+            api_val_post_disassociation = target_sat.api.Host().search(
+                query={"search": f'name={vm_name}'}
+            )[0]
```

Comment:
I don't see a strong need to specify pre- and post-disassociation of the host here, as the code remains clear and the host remains a host throughout. I'd recommend simplifying this for better readability. 
```suggestion
            host = target_sat.api.Host().search(
                query={"search": f'name={vm_name}'}
            )[0]
```

Reply:
I second @Gauravtalreja1 comment. Using just `host` variable is IMO better readable then `api_val_pre_disassociation`/`api_val_post_disassociation`. 
1. The disassociation process is clear from the test flow.
2. You don't compare the pre/post values, so no need for two long names.